server/feed.js

```diff
@@ -0,0 +1,49 @@
+var soundcloud = require('../soundcloud/api-client');
+var knexfile = require('../knexfile');
+var knex = require('knex')(knexfile);
+var _ = require('lodash');
+var Promise = require('bluebird');
+
+module.exports = function fetchFeedApi(soundcloudUserId, token, nextLink) {
+    return Promise.join(
+        fetchSoundcloudFeed(token, nextLink),
+        fetchBlacklist(soundcloudUserId),
+        function (feed, blacklist) {
+            feed.tracks = _.filter(feed.tracks, function (track) {
+                return typeof track !== 'undefined' && !_.includes(blacklist, track.id);
```

Comment:
I hope it's not, couldn't figure out why I get some undefined value here. Will investigate.